### DISCUSSION TITLE
Use overmind if available

### DISCUSSION THREAD

**Discussion by MatheusRich** [thoughtbot/suspenders#1305] [Ideas]
Use overmind instead of foreman, if available

[Overmind] is a process manager (like foreman) for Procfile-based
applications. The advantage is that it [allows to use debuggers] like
pry or `binding.irb`.

[Overmind]: https://github.com/DarthSim/overmind
[allows to use debuggers]: https://heyvaleria.github.io/coding/2022/04/27/overmind.html

(Maybe this should be a PR against Rails?)

**Comment by stevepolitodesign** [comment]
> Use overmind instead of foreman, if available

Meaning, we'd check to see if Overmind is available on the user's system? 

Currently, Rails generates [./bin/dev](https://github.com/rails/jsbundling-rails/blob/main/lib/install/dev) for applications using [jsbundling-rails](https://github.com/rails/jsbundling-rails).

We could probably modify that file to check for the existence of Overmind.

**Comment by MatheusRich** [comment]
@stevepolitodesign yes! I forgot to attach the code for it:

```sh
#!/usr/bin/env sh

# Magic happens here
if command -v overmind &> /dev/null
then
  overmind start -f Procfile.dev "$@"
  exit $?
fi

if ! gem list foreman -i --silent; then
  echo "Installing foreman..."
  gem install foreman
fi

exec foreman start -f Procfile.dev "$@"
```
